cryptus-app/lib/update_nfts_for_user.ts

```diff
@@ -28,24 +34,64 @@ export default async function updateNftsForUser(
 
     nfts_raw = data.assets;
   } catch (err) {
-    console.error(err);
-    console.log("response = ", res);
+    // console.error(err);
+    // console.log("response = ", res);
     return null;
   }
 
   // console.log("traits = ", nfts_raw[0]);
-
   const nft_clean = nfts_raw.map((nft) => {
-    const properties = JSON.stringify(
-      nft.traits.map((trait) => {
-        return {
+    // Sort properties here
+    const properties: traits[] = [];
+    if (!(nft.traits === undefined) && !(nft.traits.length == 0)) {
+      const first_trait: traits = {
+        name: nft.traits[0].trait_type,
+        value: nft.traits[0].value,
+        count: nft.traits[0].trait_count,
+        rarity: nft.traits[0].trait_count / 2010863, //  CryptoKitties (CK)
+      };
+      properties.push(first_trait);
+
+      for (const trait of nft.traits) {
+        const new_trait: traits = {
           name: trait.trait_type,
           value: trait.value,
           count: trait.trait_count,
-          rarity: trait.trait_count / 10000, //TODO find the total number of nfts in a collection
+          //TODO find the total number of nfts in a collection instead of a const
+          // rarity: trait.trait_count / 10000,
+          rarity: trait.trait_count / 2010863, //  CryptoKitties (CK)
         };
-      })
-    );
+        if (
+          !(
+            new_trait.name in
+            properties.map((prop) => {
+              return prop.name;
+            })
+          )
+        ) {
+          properties_loop: for (
+            let index = 0;
+            index < properties.length;
+            index++
+          ) {
+            if (new_trait.rarity < properties[index].rarity) {
+              properties.splice(index, 0, new_trait);
+              break properties_loop;
+            }
+          }
+        }
+      }
+    }
+
+    // console.log(properties);
+    // const nft_properties = JSON.stringify(properties);
+    const nft_properties = properties;
+    let rarity_rank: number = 0;
+    // Implement something in case there is no trait
+    if (properties.length > 0) {
+      rarity_rank = properties[0].rarity * 2010863; //  CryptoKitties (CK)-- Algo is only based on it's rarest trait
```

Comment:
where is the actual rarity_rank calculated? from what I can see it's only mocked